### PR TITLE
SALTO-1933: added support deploying project components

### PR DESCRIPTION
**Pull request by alonstern**
Moved project component to a standalone field, removed redundant values in it, and made it deployable

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/src/config.ts [hunk 1]
```diff
@@ -768,14 +773,42 @@ const DEFAULT_TYPE_CUSTOMIZATIONS: JiraApiConfig['types'] = {
...
+        { fieldName: 'componentBean', fieldType: 'ProjectComponent' },
+      ],
+      fieldsToHide: [
+        { fieldName: 'id' },
+      ],
       fieldsToOmit: [
         { fieldName: 'issueCount' },
-        { fieldName: 'id' },
         { fieldName: 'projectId' },
+        { fieldName: 'project' },
```
Comment (netama) [inline packages/jira-adapter/src/config.ts:790]: how come we're both overriding this field's type and omitting it? it also looks like we're overriding it below.
if this is not the final form, I'm not sure we should use the infra for it (but I might be missing something about the use case)

File: packages/jira-adapter/src/filters/project.ts [hunk 2]
```diff
@@ -108,9 +109,13 @@ const filter: FilterCreator = ({ config, client }) => ({
           change,
           client,
           apiDefinitions: config.apiDefinitions,
-          fieldsToIgnore: ['workflowScheme', 'issueTypeScreenScheme', 'fieldConfigurationScheme', 'issueTypeScheme'],
+          fieldsToIgnore: isModificationChange(change)
+            ? ['components', 'workflowScheme', 'issueTypeScreenScheme', 'fieldConfigurationScheme', 'issueTypeScheme']
```
Comment (netama) [inline packages/jira-adapter/src/filters/project.ts:113]: nit: at least some of these seem to appear in a lot of places (e.g. `'issueTypeScheme'`), let's move the recurring ones to constants?

File: packages/jira-adapter/src/filters/project_component.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            client,
+            apiDefinitions: config.apiDefinitions,
+          })
+        } catch (err) {
+          if (err instanceof clientUtils.HTTPError
+            && Array.isArray(err.response.data.errorMessages)
+            && err.response.data.errorMessages
+              .includes(`The component with id ${getChangeData(change).value.id} does not exist.`)
+          ) {
+            return
```
Comment (netama) [inline packages/jira-adapter/src/filters/project_component.ts:76]: maybe add a log?

**Review by netama — APPROVED** [review]
pre-approving to unblock, but might be worth discussing the balance between config and code here if we're changing the same field value in multiple steps

File: packages/jira-adapter/src/config.ts (hunk 1 above)
Comment (alonstern) [reply packages/jira-adapter/src/config.ts:790]: the type override was redundant, removed

**Pull request merged**